calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -216,6 +219,123 @@ def test_validate_arguments(self, case, sys_exit_called):
             self.assertEqual(m_sys_exit.called, sys_exit_called)
 
 
+    @patch('calico_ctl.node.os', autospec=True)
+    @patch('calico_ctl.node.check_system', autospec=True)
+    @patch('calico_ctl.node.get_host_ips', autospec=True)
+    @patch('calico_ctl.node.warn_if_unknown_ip', autospec=True)
+    @patch('calico_ctl.node.warn_if_hostname_conflict', autospec=True)
+    @patch('calico_ctl.node.install_kubernetes', autospec=True)
+    @patch('calico_ctl.node.client', autospec=True)
+    @patch('calico_ctl.node.docker_client', autospec=True)
+    @patch('calico_ctl.node.docker', autospec=True)
+    @patch('calico_ctl.node._find_or_pull_node_image', autospec=True)
+    def test_node_start(self, m_find_or_pull_node_image, m_docker,
+                        m_docker_client, m_client, m_install_kube,
+                        m_warn_if_hostname_conflict, m_warn_if_unknown_ip,
+                        m_get_host_ips, m_check_system, m_os):
+        """
+        Test node_start function in calicoctl node command
+        """
+        # Set up mock objects
+        m_os.path.exists.return_value = False
+        m_get_host_ips.return_value = ['1.1.1.1', '2.2.2.2', '3.3.3.3']
+        m_os.getenv.return_value = '1.1.1.1:80'
+        m_docker.utils.create_host_config.return_value = 'host_config'
+        m_docker_client.create_container.return_value = {'Id':666}
```

Comment:
Looks like you're probably missing lines 240-242 also.  Might want to set a return value on `os.getenv` and mock out `sys.exit`